### ISSUE TITLE
[Bug]: tpch q2 quries is hung when table region is empty on 2*cn mo cluster

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#7853]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):5c7ea0b2
- Hardware parameters: aws m6a.4xl (16C 64G)
- OS type:
- Others:
```

### Actual Behavior

For tpch 1g queries, if table region is empty for some reason, q2 quires will be hung all the time on 2 cn topo.

1 cn :
mysql> select         s_acctbal,         s_name,         n_name,         p_partkey,         p_mfgr,         s_address,         s_phone,         s_comment from         part,         supplier,         partsupp,         nation,         region where         p_partkey = ps_partkey         and s_suppkey = ps_suppkey         and p_size = 48         and p_type like '%TIN'         and s_nationkey = n_nationkey         and n_regionkey = r_regionkey         and r_name = 'MIDDLE EAST'         and ps_supplycost = (                 select                         min(ps_supplycost)                 from
    ->              partsupp,                         supplier,                         nation,                         region                 where                         p_partkey = ps_partkey
    ->            and s_suppkey = ps_suppkey                         and s_nationkey = n_nationkey                         and n_regionkey = r_regionkey                         and r_name = 'MIDDLE EAST'
    ->    ) order by         s_acctbal desc,         n_name,         s_name,         p_partkey limit 100;
Empty set (0.71 sec)

mysql> select         s_acctbal,         s_name,         n_name,         p_partkey,         p_mfgr,         s_address,         s_phone,         s_comment from         part,         supplier,         partsupp,         nation,         region where         p_partkey = ps_partkey         and s_suppkey = ps_suppkey         and p_size = 48         and p_type like '%TIN'         and s_nationkey = n_nationkey         and n_regionkey = r_regionkey         and r_name = 'MIDDLE EAST'         and ps_supplycost = (                 select                         min(ps_supplycost)                 from
  partsupp,                         supplier,                         nation,                         region                 where                         p_partkey = ps_partkey            and s_suppkey = ps_suppkey                         and s_nationkey = n_nationkey                         and n_regionkey = r_regionkey                         and r_name = 'MIDDLE EAST'    ) order by         s_acctbal desc,         n_name,         s_name,         p_partkey limit 100;
Empty set (0.19 sec)

2 cn:
mysql> select         s_acctbal,         s_name,         n_name,         p_partkey,         p_mfgr,         s_address,         s_phone,         s_comment from         part,         supplier,         partsupp,         nation,         region where         p_partkey = ps_partkey         and s_suppkey = ps_suppkey         and p_size = 48         and p_type like '%TIN'         and s_nationkey = n_nationkey         and n_regionkey = r_regionkey         and r_name = 'MIDDLE EAST'         and ps_supplycost = (                 select                         min(ps_supplycost)                 from
  partsupp,                         supplier,                         nation,                         region                 where                         p_partkey = ps_partkey            and s_suppkey = ps_suppkey                         and s_nationkey = n_nationkey                         and n_regionkey = r_regionkey                         and r_name = 'MIDDLE EAST'    ) order by         s_acctbal desc,         n_name,         s_name,         p_partkey limit 100;
Empty set (0.19 sec)

mysql> select         s_acctbal,         s_name,         n_name,         p_partkey,         p_mfgr,         s_address,         s_phone,         s_comment from         part,         supplier,         partsupp,         nation,         region where         p_partkey = ps_partkey         and s_suppkey = ps_suppkey         and p_size = 48         and p_type like '%TIN'         and s_nationkey = n_nationkey         and n_regionkey = r_regionkey         and r_name = 'MIDDLE EAST'         and ps_supplycost = (                 select                         min(ps_supplycost)                 from
  partsupp,                         supplier,                         nation,                         region                 where                         p_partkey = ps_partkey            and s_suppkey = ps_suppkey                         and s_nationkey = n_nationkey                         and n_regionkey = r_regionkey                         and r_name = 'MIDDLE EAST'    ) order by         s_acctbal desc,         n_name,         s_name,         p_partkey limit 100;

^C^C -- query aborted
ERROR 20101 (HY000): internal error: pipeline closed unexpectedly

**Comment by dongdongyang33** [comment]
checking

**Comment by aressu1985** [comment]
fixed:4410d861


mysql> delete from region;
Query OK, 5 rows affected (0.11 sec)

mysql> select * from region;
Empty set (0.01 sec)

mysql> select s_acctbal, s_name, n_name, p_partkey, p_mfgr, s_address, s_phone, s_comment from part, supplier, partsupp, nation, region where p_partkey = ps_partkey and s_suppkey = ps_suppkey and p_size = 48 and p_type like '%TIN' and s_nationkey = n_nationkey and n_regionkey = r_regionkey and r_name = 'MIDDLE EAST' and ps_supplycost = ( select min(ps_supplycost) from
    -> partsupp, supplier, nation, region where p_partkey = ps_partkey and s_suppkey = ps_suppkey and s_nationkey = n_nationkey and n_regionkey = r_regionkey and r_name = 'MIDDLE EAST' ) order by s_acctbal desc, n_name, s_name, p_partkey limit 100;
Empty set (0.38 sec)

mysql>
